h/services/user_expunge.py

```diff
@@ -0,0 +1,233 @@
+import logging
+
+from sqlalchemy import and_, delete, func, select, update
+
+from h.models import (
+    Annotation,
+    AuthTicket,
+    FeatureCohortUser,
+    Flag,
+    Group,
+    GroupMembership,
+    Job,
+    Token,
+    User,
+)
+
+log = logging.getLogger(__name__)
+
+
+class UserExpungeService:
```

Comment:
There's already an existing [`UserDeleteService`](https://github.com/hypothesis/h/blob/8453ab53aedce716c184f387f1d2ea6ac50a24a6/h/services/user_delete.py#L7-L48) but the way it deletes users is terrible. My plan is to replace it with this new `UserExpungeService`. I may just replace the body of the existing `UserDeleteService` though, rather than add a new service with a different name.